pkg/kubelet/network/plugins.go

```diff
@@ -54,6 +60,9 @@ type NetworkPlugin interface {
 	// for a plugin by name, e.g.
 	Name() string
 
+	// Returns a bitfield of plugin capabilities
+	Capabilities() uint
```

Comment:
A String Set is what I meant. An Int Set might be better though for this
use case.

On Fri, Mar 11, 2016 at 9:55 AM, Dan Williams <notifications@github.com>
wrote:

> In pkg/kubelet/network/plugins.go
> <https://github.com/kubernetes/kubernetes/pull/22827#discussion_r55863526>
> :
>
> > @@ -54,6 +60,9 @@ type NetworkPlugin interface {
> >  	// for a plugin by name, e.g.
> >  	Name() string
> >
> > +	// Returns a bitfield of plugin capabilities
> > +	Capabilities() uint
>
> nit: I prefer returning a Set instead of uint.
>
> Do you mean an Int Set or a String Set? I only get one hit for Int sets
> (pkg/client/restclient/urlbackoff.go) so it seems like there aren't a ton
> of Int/Int64/Byte Set users yet. I'm OK moving to a Set though if you want
> one.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/22827/files#r55863526>.
>